docs/how-to/enforce-permissions/data-filtering.mdx

```diff
@@ -0,0 +1,67 @@
+---
+sidebar_position: 4
+title: Data Filtering
+---
+
+Data filtering in permissions represents is a different kind of approach when enforcing permissions.
+Instead of merely granting or denying access, it curates what users see, tailoring the data to their individual permissions.
+This method ensures not only secure access but also optimized data delivery.
+
+## Simple usage
+
+```go
+package main
+
+import (
+	"fmt"
+)
+import p "github.com/permitio/permit-golang/pkg/permit"
+import "github.com/permitio/permit-golang/pkg/config"
+import "github.com/permitio/permit-golang/pkg/enforcement"
+
+func main() {
+	// Create permit client
+	permitConfig := config.NewConfigBuilder("").Build()
+	permit := p.New(permitConfig)
+
+	requestContext := map[string]string{
+		"source": "docs",
+	}
+	user := enforcement.UserBuilder("john@doe.com").Build()
+	var action enforcement.Action = "read"
+	resourcesToCheck := []enforcement.ResourceI{
+		enforcement.ResourceBuilder("document").WithID("document-1").WithTenant(enforcement.DefaultTenant),
+		enforcement.ResourceBuilder("folder").WithID("folder-1").WithTenant(enforcement.DefaultTenant),
+		enforcement.ResourceBuilder("document").WithID("document-2").WithTenant(enforcement.DefaultTenant),
+		enforcement.ResourceBuilder("document").WithID("document-3").WithTenant(enforcement.DefaultTenant),
+	}
+	var allowedResources []enforcement.ResourceI
+	var err error
+	// Filter the objects
+	allowedResources, err = permit.FilterObjects(user, action, requestContext, resourcesToCheck...)
+	if err != nil {
+		fmt.Printf("Error enforcing permissions: %s", err)
+	} else {
+		// The response indexes correspond to the request indexes
+		for i, resource := range allowedResources {
+			fmt.Printf("%d. User '%s' is PERMITTED to '%s' a '%s' with id '%s'\n",
+				i, user.Key, action, resource.GetType(), resource.GetID(),
+			)
+		}
+	}
+}
+
+```
+
+## Advanced Usage
+
+:::tip
+
+Stay Tuned !
```

Comment:
Feel like this should really mention the term "partial evaluation"

Reply:
fixed